### DISCUSSION TITLE
New software version for CF editor interface (and XML improvements)

### DISCUSSION THREAD

**Discussion by efisher008** [cf-convention/discuss#391] [Announcements relevant to CF]
Dear all,

The Standard Names Committee are pleased to announce that the CF editor has been updated to a new interface. This update has been combined with a number of open issues pertaining to the editor's functionality, which are described below. The editor is still accessible at the existing link: https://cfeditor.ceda.ac.uk/proposals/1

The update to the editor automates changes to the table XML that were agreed in the following issues:
https://github.com/cf-convention/cf-conventions/issues/367 - No GRIB or AMIP tags
https://github.com/cf-convention/cf-conventions/issues/500 - Additional header tag for CF-conventions string
https://github.com/cf-convention/cf-conventions/issues/511 - Additional header tag for first published date
https://github.com/cf-convention/cf-conventions/issues/509 - Allow alias id tags to contain more than one standard name entry_id reference

These changes were made in the most recent versions of the Standard Names table by hand editing the XML, so having these now automated in the editor is a significant improvement. The new software version has now been deployed and was live as of 1st November 2024. 

If there are any issues or queries with the performance of the CF editor, please let us know in this discussion, and @japamment, @feggleton or I will look into it. 

Best regards,
Ellie Fisher, Alison Pamment, Francesca Eggleton

**Comment by larsbarring** [comment]
Hi Ellie and Standard Names Moderators,

This is really excellent news --- thanks a lot to you, and to the CF Editor developers behind the scene!  :-) 

A couple of minor suggestions regarding xml file layout/style: 
*  [This comment](https://github.com/cf-convention/cf-convention.github.io/issues/457#issuecomment-2011214628)  suggests that the aliases should be [alphabetically] sorted. 
*  Currently there is a extra empty line in each alias entry, e.g.
```xml
  <alias id="chlorophyll_concentration_in_sea_water">
    <entry_id>mass_concentration_of_chlorophyll_in_sea_water</entry_id>
     
  </alias>
```

While neither of these two are essential for the xml file (format) as such, I think that it would be useful --- and easy --- to update the CF Editor to implement sorting (on the alias id) and to remove the extra empty line.

As I have absolutely no insight into the inner workings of the CF Editor, the following ideas are as much questions as suggestions:

*  Is the produced xml file validated against the xls schema file? If not, adding such a functionality would catch many/almost all typos of the kind we have seen in the past.
* Would it be possible to include in the CF Editor validation of the canonical units strings using the [cfunits](/https://ncas-cms.github.io/cfunits/) package. If so, any malformed (or otherwise "odd") units would be caught.

**Reply by larsbarring** [reply]
A further suggestion following on from https://github.com/cf-convention/vocabularies/issues/238: I believe it would be easy to add a check that there are no **`/`** in the units string.

**Reply by efisher008** [reply]
Thank you very much @larsbarring for your comments. I have added these comments as issue tickets to the development area for the vocab editor. It was considered lower priority for this deployment compared to the major changes to the XML listed in the original post above, but will definitely be revisited when the editor software is next updated!

I would suggest @japamment might have an idea of your other questions:

> Is the produced xml file validated against the xls schema file? If not, adding such a functionality would catch many/almost all typos of the kind we have seen in the past.

> Would it be possible to include in the CF Editor validation of the canonical units strings using the [cfunits](https://github.com/https://ncas-cms.github.io/cfunits/) package. If so, any malformed (or otherwise "odd") units would be caught.
